Windows builds are broken

```
C:\projects\osrm\osrm-deps\boost\include\boost-1_59\boost/mpl/eval_if.hpp(41): error C2504: 'boost::result_of<const UnaryFunc (unsigned int &)>': base class undefined [C:\projects\osrm\build\ENGINE.vcxproj]
```

This was broken by #3015.

--- a/features/bicycle/surface.feature
+++ b/features/bicycle/surface.feature
@@ -8,7 +8,7 @@ Feature: Bike - Surfaces
         Then routability should be
             | highway  | surface               | bothw |
             | cycleway |                       | 48 s  |
-            | cycleway | asphalt               | 48 s  |
+            | cycleway | asphalt               | 47.9 s|
             | cycleway | cobblestone:flattened | 72 s  |
             | cycleway | paving_stones         | 72 s  |
             | cycleway | compacted             | 72 s  |
@@ -26,7 +26,7 @@ Feature: Bike - Surfaces
         Then routability should be
         | highway  | surface | bothw |
         | cycleway |         | 48 s  |
-        | path     |         | 59.9 s +- 1 |
+        | path     |         | 59.9 s|
         | track    |         | 60 s  |
         | track    | asphalt | 60 s  |
         | path     | asphalt | 60 s  |

--- a/features/bicycle/surface.feature
+++ b/features/bicycle/surface.feature
@@ -26,7 +26,7 @@ Feature: Bike - Surfaces
         Then routability should be
         | highway  | surface | bothw |
         | cycleway |         | 48 s  |
-        | path     |         | 60 s  |
+        | path     |         | 59.9 s +- 1 |
         | track    |         | 60 s  |
         | track    | asphalt | 60 s  |
         | path     | asphalt | 60 s  |

--- a/features/step_definitions/routability.js
+++ b/features/step_definitions/routability.js
@@ -37,7 +37,7 @@ module.exports = function () {
                                 outputRow[direction] = result[direction].status ?
                                     result[direction].status.toString() : '';
                                 break;
-                            case /^\d+ s/.test(want):
+                            case /^[\d\.]+ s/.test(want):
                                 // the result here can come back as a non-number value like
                                 // `diff`, but we only want to apply the unit when it comes
                                 // back as a number, for tableDiff's literal comparison

--- a/features/step_definitions/routability.js
+++ b/features/step_definitions/routability.js
@@ -127,23 +127,6 @@ module.exports = function () {
                 var parseRes = (key, scb) => {
                     if (result.forw[key] === result.backw[key]) {
                         result.bothw[key] = result.forw[key];
-                    // FIXME these time and speed checks are stopgaps for precision errors in how
-                    // OSRM returns inconsistent durations for rev/for requests along the same way
-                    } else if (key === 'time') {
-                        var range = [result.forw[key] - 1, result.forw[key] + 1];
-                        if (result.backw[key] >= range[0] && result.backw[key] <= range[1])
-                            // usually when we see minor differences here there's an integer
-                            // duration value and one that comes back with a .9 or .1 rounding.
-                            // This returns the integer one
-                            result.bothw[key] = parseInt(result.forw[key]) === result.forw[key] ? result.forw[key] : result.backw[key];
-                        else
-                            result.bothw[key] = 'diff';
-                    } else if (key === 'speed') {
-                        if (Math.abs(result.backw.time - result.forw.time) < 0.2) {
-                            result.bothw[key] = parseInt(result.forw[key]) === result.forw[key] ? result.forw[key] : result.backw[key];
-                        } else {
-                            result.bothw[key] = 'diff';
-                        }
                     } else {
                         result.bothw[key] = 'diff';
                     }

--- a/include/engine/geospatial_query.hpp
+++ b/include/engine/geospatial_query.hpp
@@ -400,7 +400,8 @@ template <typename RTreeT, typename DataFacadeT> class GeospatialQuery
         }
         if (data.reverse_segment_id.id != SPECIAL_SEGMENTID)
         {
-            reverse_weight *= 1.0 - ratio;
+            const EdgeWeight difference = reverse_weight * ratio;
+            reverse_weight -= difference;
         }
 
         auto transformed = PhantomNodeWithDistance{PhantomNode{data,

